blockstore/splitstore/splitstore.go

```diff
@@ -618,13 +620,27 @@ func (s *SplitStore) trackTxnRefMany(cids []cid.Cid) error {
 	return nil
 }
 
+// This could get _real_ slow (many minutes to hours) and will block tipset verification. Can we do this lazily and/or force
+// the GC thread to handle this?
+//
+// I.e., take a lock to make sure GC is paused, then stick this in a "to traverse" queue. Then, when
+// GC "resumes", it first checks that queue.
```

Comment:
yeah, I have been thinking about this too.

It hasn't been observed in testing, but then again it's about 30 compactions so far and it will probably happen very infrequently.

Reply:
Implemented concurrent marking -- see https://github.com/filecoin-project/lotus/pull/6474/commits/da0feb3fa43a78bd7be3d8aa49495cd31e1a87c2

Reply:
and it works pretty well -- there is a compaction running right now, and it cleans up txn refs in at most two iterations with concurrent marking.